File: week5/dinosaur.rb
class Dinosaur
  def initialize( name )
    @name = name
    puts "I am #{@name}, hear me ROAR!"
  end
end

my_dino = Dinosaur.new("Triceratops")


#### User

class User
  # attr_reader :first # getter
  # attr_writer :last # setter
  attr_accessor :first, :last
  @@population = 0
  def initialize( first, last )
    @first = first
    @last = last
    @@population += 1
  end
  ## getter for first
  # def first   
  #   @first
  # end
  ## setter for first
  # def first=(first)
  #   @first = first
  # end 
  # def last
  #   @last
  # end
  # def last=(last)
  #   @last = last
  # end
  def full_name
    "#{@first} #{@last}"
  end
  def self.population
    @@population
  end
end

$person1 = User.new("Cam","Crews")
$person2 = User.new("Mikey","Lane")

puts person2.full_name

File: week5/dinosaur.js
// ES5 version
function Dinosaur(name){
  this.name = name;
  console.log(`I am ${name} hear me ROAR!`);
}

var myDino = new Dinosaur("Triceratops");



// User
function User(first, last){
  User.population = (User.population || 0) + 1;
  this.first = first;
  this.last = last;
}
User.prototype.fullName = function(){
  return `${this.first} ${this.last}`
};


var person1 = new User("Cam","Crews");
var person2 = new User("Mikey","Lane");


console.log( person2.fullName() );











// ES6 class version
class Dinosaur{
  constructor(){
    console.log("ROAR");
  }

}